tests/e2e/registration-by-sms.js

```diff
@@ -3,6 +3,20 @@ const utils = require('../utils'),
       helper = require('../helper'),
       moment = require('moment');
 
+const computeExpectedDate = async () => {
+  const reportedDate = await browser.executeAsyncScript(async () => {
+    const callback = arguments[arguments.length - 1];
+    const element = document.querySelector('.detail .relative-date-content');
+    const attribute = element.getAttribute('data-date-options');
+    callback(JSON.parse(attribute));
+  });
+
+  const start = moment(reportedDate.date).startOf('day').subtract(12, 'weeks');
+  const expectedDate = start.add(40, 'weeks');
+
+  return expectedDate;
```

Comment:
I think there's a more Protractor like way of doing this which would be a little simpler, eg:

```js
const reportedDate = element(by.css('.detail .relative-date-content')).getAttribute('data-record-id');
```

This way you don't need to executeAsyncScript or do any async/await - protractor does that all for you.